src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/ToValueTask/ToValueTask.cs

```diff
@@ -0,0 +1,13 @@
+#nullable enable
+
+using System.Threading.Tasks;
+
+namespace System
+{
+    partial struct AsyncPipeline<T>
+    {
+        public ValueTask<T> ToValueTask()
+            =>
+            isCanceled is false ? valueTask : ValueTask.FromCanceled<T>(CanceledToken());
```

Comment:
Or we can just rename `ToValueTask()` to `AsValueTask()` - without any wrapping.
But in this case the implicit conversion should be removed (because `As` is not a conversion).

Reply:
Decided to keep as is.